src/Build/BackEnd/Components/RequestBuilder/RequestBuilder.cs

```diff
@@ -362,12 +362,14 @@ public async Task<BuildResult[]> BuildProjects(string[] projectFiles, PropertyDi
                 BuildRequestData data = new BuildRequestData(projectFiles[i], properties[i].ToDictionary(), explicitToolsVersion, targets, null);
 
                 BuildRequestConfiguration config = new BuildRequestConfiguration(data, _componentHost.BuildParameters.DefaultToolsVersion);
-
+                ProjectIsolationMode isolateProjects = _componentHost.BuildParameters.ProjectIsolationMode;
+                bool skipStaticGraphIsolationConstraints = (isolateProjects != ProjectIsolationMode.False && _requestEntry.RequestConfiguration.ShouldSkipIsolationConstraintsForReference(config.ProjectFullPath))
+                    || isolateProjects == ProjectIsolationMode.MessageUponIsolationViolation;
```

Comment:
This feels a little weird to me. Why should MessageUponIsolationViolation be treated differently than True here?

Reply:
With `ProjectIsolationMode.True`, the only way for `skipStaticGraphIsolationConstraints` to be `true` is if the reference is exempt via the `GraphIsolationExemptReference` item: https://github.com/dotnet/msbuild/blob/a59d7a533c9154e8aa99b823625e7eff199ddf1a/documentation/specs/static-graph-implementation-details.md#how-isolation-exemption-complicates-everything

With `ProjectIsolationMode.MessageUponIsolationViolation`, the `GraphIsolationExemptReference` item doesn't need to exist since exemptions are allowed as needed.